packages/rust/armonik/src/client/sessions.rs

```diff
@@ -0,0 +1,546 @@
+        },
+        TaskOptions,
+    },
+    utils::IntoCollection,
+};
+
+use super::GrpcCall;
+
+/// Service for handling sessions
+#[derive(Clone)]
+pub struct SessionsClient<T> {
+    inner: v3::sessions::sessions_client::SessionsClient<T>,
+}
+
+impl<T> SessionsClient<T>
+where
+    T: tonic::client::GrpcService<tonic::body::BoxBody>,
+    T::Error: Into<tonic::codegen::StdError>,
+    T::ResponseBody: tonic::codegen::Body<Data = tonic::codegen::Bytes> + Send + 'static,
+    <T::ResponseBody as tonic::codegen::Body>::Error: Into<tonic::codegen::StdError> + Send,
+{
+    pub fn with_channel(channel: T) -> Self {
+        Self {
+            inner: v3::sessions::sessions_client::SessionsClient::new(channel),
+        }
+    }
+
+    /// Get a sessions list using pagination, filters and sorting.
+    pub async fn list(
+        &mut self,
+        request: list::Request,
+    ) -> Result<list::Response, super::RequestError> {
+        self.call(request).await
+    }
+
+    /// Get a session by its id.
+    pub async fn get(&mut self, session_id: impl Into<String>) -> Result<Raw, super::RequestError> {
+        Ok(self
+            .call(get::Request {
+                id: session_id.into(),
+            })
+            .await?
+            .session)
+    }
+
+    /// Cancel a session by its id.
+    pub async fn cancel(
+        &mut self,
+        session_id: impl Into<String>,
+    ) -> Result<Raw, super::RequestError> {
+        Ok(self
+            .call(cancel::Request {
+                id: session_id.into(),
+            })
+            .await?
+            .session)
+    }
+
+    /// Create a session.
+    pub async fn create(
+        &mut self,
+        partitions: impl IntoIterator<Item = impl Into<String>>,
+        task_options: TaskOptions,
+    ) -> Result<String, super::RequestError> {
+        Ok(self
+            .call(create::Request {
+                default_task_option: task_options,
+                partition_ids: partitions.into_collect(),
+            })
+            .await?
+            .session_id)
+    }
+
+    /// Pause a session by its id.
+    pub async fn pause(
+        &mut self,
+        session_id: impl Into<String>,
+    ) -> Result<Raw, super::RequestError> {
+        Ok(self
+            .call(pause::Request {
+                id: session_id.into(),
+            })
+            .await?
+            .session)
+    }
+
+    /// Resume a paused session by its id.
+    pub async fn resume(
+        &mut self,
+        session_id: impl Into<String>,
+    ) -> Result<Raw, super::RequestError> {
+        Ok(self
+            .call(resume::Request {
+                id: session_id.into(),
+            })
+            .await?
+            .session)
+    }
+
+    /// Close a session by its id.
+    pub async fn close(
+        &mut self,
+        session_id: impl Into<String>,
+    ) -> Result<Raw, super::RequestError> {
```

Comment:
do we keep Raw or RawSession ?

Reply:
The idea was to always use `session::Raw` on the user side. Here, we are in the session module, so it's fine.